WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java

```diff
@@ -211,8 +211,10 @@ public void onClick(View v) {
                 if (!mAccountStore.hasAccessToken() && selectedSite != null && selectedSite.isJetpackConnected()) {
                     // If the user is not connected to WordPress.com, ask him to connect first.
                     startWPComLoginForJetpackStats();
-                } else {
+                } else if (selectedSite != null && selectedSite.isJetpackInstalled() && selectedSite.isJetpackConnected()){
                     ActivityLauncher.viewBlogStats(getActivity(), selectedSite);
+                } else {
+                    ActivityLauncher.startJetpackConnectionFlow(getActivity(), selectedSite);
```

Comment:
Both the `if` statements above check for a null `selectedSite`. Should that be done here, too?

Reply:
I have the null checks there to never have NPEs, not sure if null site is a valid state. I can totally add the check